firedrake/adjoint_utils/function.py

```diff
@@ -225,23 +225,32 @@ def _ad_convert_riesz(self, value, options=None):
         from firedrake import Function, Cofunction
 
         options = {} if options is None else options
-        riesz_representation = options.get("riesz_representation", "l2")
+        riesz_representation = options.get("riesz_representation", "L2")
         solver_options = options.get("solver_options", {})
         V = options.get("function_space", self.function_space())
 
-        if riesz_representation != "l2" and not isinstance(value, Cofunction):
-            raise TypeError("Expected a Cofunction")
-        elif not isinstance(value, (Number, Cofunction, Function)):
+        if not isinstance(value, (Number, Cofunction, Function, Number)):
```

Comment:
Change into 
`if not isinstance(value, (Cofunction, Function, Number)):`
to remove redundant `Number`

Reply:
With the latest commits, `_ad_convert_riesz` should not allow `value` to be a `Number` when `value !=0`. So, the first line of `_ad_convert_riesz` function verifies if `value == 0.`. So, at this point, values must be either a `Cofunction` or a `Function`:
```
if not isinstance(value, (Cofunction, Function)):
    raise TypeError("Expected a Cofunction or a Function")
```